File: wrappers/nikto.go
package main

import (
	"fmt"
	"os"
	"os/exec"
	"path/filepath"
	"strings"
)

// تعريف الألوان
const (
	RED     = "\033[0;31m"
	GREEN   = "\033[0;32m"
	YELLOW  = "\033[0;33m"
	BLUE    = "\033[0;34m"
	MAGENTA = "\033[0;35m"
	CYAN    = "\033[0;36m"
	WHITE   = "\033[1;37m"
	BOLD    = "\033[1m"
	RESET   = "\033[0m"
)

// عرض الشعار
func showBanner() {
	fmt.Printf("%s%s", BOLD, BLUE)
	fmt.Println("  _   _ _ _    _        ")
	fmt.Println(" | \\ | (_) | _| |_ ___  ")
	fmt.Println(" |  \\| | | |/ / __/ _ \\ ")
	fmt.Println(" | |\\  | |   <| || (_) |")
	fmt.Println(" |_| \\_|_|_|\\_\\\\__\\___/ ")
	fmt.Printf("%s%s====================\n", RESET, CYAN)
	fmt.Printf("%sNikto Web Scanner - v3.0.0%s\n", YELLOW, RESET)
	fmt.Printf("%s====================%s\n", CYAN, RESET)
	fmt.Printf("%sGo Wrapper%s\n\n", GREEN, RESET)
}

// عرض المساعدة
func showHelp() {
	fmt.Printf("%sاستخدام:%s %s [خيارات]\n\n", BOLD, RESET, os.Args[0])
	fmt.Printf("%sالخيارات الأساسية:%s\n", BOLD, RESET)
	fmt.Printf("  %s-h, --host%s HOST       الهدف المراد فحصه\n", GREEN, RESET)
	fmt.Printf("  %s-p, --port%s PORT       المنفذ المراد فحصه (الافتراضي: 80)\n", GREEN, RESET)
	fmt.Printf("  %s-o, --output%s FILE     حفظ النتائج في ملف\n", GREEN, RESET)
	fmt.Printf("  %s-F, --Format%s FORMAT   تنسيق الملف (csv, htm, xml, json)\n", GREEN, RESET)
	fmt.Printf("  %s-D, --Display%s OPTION  خيارات العرض (1-4,V,E)\n", GREEN, RESET)
	fmt.Printf("  %s--help%s                عرض هذه المساعدة\n\n", GREEN, RESET)
	fmt.Printf("%sأمثلة:%s\n", BOLD, RESET)
	fmt.Printf("  %s -h example.com\n", os.Args[0])
	fmt.Printf("  %s -h example.com -p 443 -o report.html -F htm\n\n", os.Args[0])
	fmt.Printf("%sملاحظة:%s هذا الغلاف يمرر جميع المعلمات إلى nikto.pl الأساسي.\n", YELLOW, RESET)
}

// التحقق من وجود Perl
func checkPerl() bool {
	cmd := exec.Command("perl", "--version")
	if err := cmd.Run(); err != nil {
		fmt.Printf("%sخطأ: Perl غير مثبت. الرجاء تثبيت Perl لتشغيل Nikto.%s\n", RED, RESET)
		return false
	}
	return true
}

// البحث عن سكربت Nikto الأساسي
func findNiktoScript() (string, bool) {
	// الحصول على المسار الحالي للسكربت
	exePath, err := os.Executable()
	if err != nil {
		fmt.Printf("%sخطأ: لا يمكن تحديد مسار البرنامج الحالي.%s\n", RED, RESET)
		return "", false
	}

	// الحصول على مسار المجلد الأب
	parentDir := filepath.Dir(filepath.Dir(exePath))
	niktoScript := filepath.Join(parentDir, "nikto.pl")

	// التحقق من وجود السكربت
	if _, err := os.Stat(niktoScript); os.IsNotExist(err) {
		fmt.Printf("%sخطأ: لم يتم العثور على سكربت Nikto الأساسي في %s%s\n", RED, niktoScript, RESET)
		return "", false
	}

	return niktoScript, true
}

// تشغيل Nikto
func runNikto(args []string) {
	// التحقق من وجود Perl
	if !checkPerl() {
		os.Exit(1)
	}

	// البحث عن سكربت Nikto
	niktoScript, found := findNiktoScript()
	if !found {
		os.Exit(1)
	}

	// تجهيز الأمر
	cmdArgs := []string{niktoScript}
	if len(args) > 1 {
		cmdArgs = append(cmdArgs, args[1:]...)
	}

	// عرض الأمر الذي سيتم تنفيذه
	fmt.Printf("%sجاري بدء الفحص...%s\n", CYAN, RESET)
	fmt.Printf("%sالأمر:%s perl %s\n\n", YELLOW, RESET, strings.Join(cmdArgs, " "))

	// تنفيذ Nikto
	cmd := exec.Command("perl", cmdArgs...)
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr

	if err := cmd.Run(); err != nil {
		fmt.Printf("\n%sحدث خطأ أثناء تنفيذ Nikto: %v%s\n", RED, err, RESET)
		os.Exit(1)
	}

	fmt.Printf("\n%sاكتمل الفحص.%s\n", GREEN, RESET)
}

func main() {
	// إذا لم يتم تمرير أي معلمات أو تم طلب المساعدة
	if len(os.Args) == 1 || contains(os.Args, "--help") {
		showBanner()
		showHelp()
		os.Exit(0)
	}

	// عرض الشعار
	showBanner()

	// تشغيل Nikto
	runNikto(os.Args)
}

// التحقق من وجود قيمة في مصفوفة
func contains(arr []string, val string) bool {
	for _, item := range arr {
		if item == val {
			return true
		}
	}
	return false
}

File: wrappers/nikto.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# Nikto Python Wrapper
# تطوير: Manus AI - 2025

import os
import sys
import argparse
import subprocess
import platform
from pathlib import Path

# تعريف الألوان
class Colors:
    RED = '\033[0;31m'
    GREEN = '\033[0;32m'
    YELLOW = '\033[0;33m'
    BLUE = '\033[0;34m'
    MAGENTA = '\033[0;35m'
    CYAN = '\033[0;36m'
    WHITE = '\033[1;37m'
    BOLD = '\033[1m'
    RESET = '\033[0m'

# شعار Nikto المحسن بالألوان
def show_banner():
    print(f"{Colors.BOLD}{Colors.BLUE}")
    print("  _   _ _ _    _        ")
    print(" | \\ | (_) | _| |_ ___  ")
    print(" |  \\| | | |/ / __/ _ \\ ")
    print(" | |\\  | |   <| || (_) |")
    print(" |_| \\_|_|_|\\_\\\\__\\___/ ")
    print(f"{Colors.RESET}{Colors.CYAN}===================={Colors.RESET}")
    print(f"{Colors.YELLOW}Nikto Web Scanner - v3.0.0{Colors.RESET}")
    print(f"{Colors.CYAN}===================={Colors.RESET}")
    print(f"{Colors.GREEN}Python Wrapper{Colors.RESET}")
    print("")

# وظيفة المساعدة
def show_help():
    print(f"{Colors.BOLD}استخدام:{Colors.RESET} python3 {sys.argv[0]} [خيارات]")
    print("")
    print(f"{Colors.BOLD}الخيارات الأساسية:{Colors.RESET}")
    print(f"  {Colors.GREEN}-h, --host{Colors.RESET} HOST       الهدف المراد فحصه")
    print(f"  {Colors.GREEN}-p, --port{Colors.RESET} PORT       المنفذ المراد فحصه (الافتراضي: 80)")
    print(f"  {Colors.GREEN}-o, --output{Colors.RESET} FILE     حفظ النتائج في ملف")
    print(f"  {Colors.GREEN}-F, --Format{Colors.RESET} FORMAT   تنسيق الملف (csv, htm, xml, json)")
    print(f"  {Colors.GREEN}-D, --Display{Colors.RESET} OPTION  خيارات العرض (1-4,V,E)")
    print(f"  {Colors.GREEN}--help{Colors.RESET}                عرض هذه المساعدة")
    print("")
    print(f"{Colors.BOLD}أمثلة:{Colors.RESET}")
    print(f"  python3 {sys.argv[0]} -h example.com")
    print(f"  python3 {sys.argv[0]} -h example.com -p 443 -o report.html -F htm")
    print("")
    print(f"{Colors.YELLOW}ملاحظة:{Colors.RESET} هذا الغلاف يمرر جميع المعلمات إلى nikto.pl الأساسي.")

def find_nikto_script():
    # الحصول على المسار الحالي للسكربت
    script_path = Path(os.path.abspath(__file__))
    nikto_dir = script_path.parent.parent
    nikto_script = nikto_dir / "nikto.pl"
    
    if not nikto_script.exists():
        print(f"{Colors.RED}خطأ: لم يتم العثور على سكربت Nikto الأساسي في {nikto_script}{Colors.RESET}")
        sys.exit(1)
    
    return str(nikto_script)

def check_perl():
    try:
        subprocess.run(["perl", "--version"], stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
    except (subprocess.SubprocessError, FileNotFoundError):
        print(f"{Colors.RED}خطأ: Perl غير مثبت. الرجاء تثبيت Perl لتشغيل Nikto.{Colors.RESET}")
        sys.exit(1)

def run_nikto(args):
    nikto_script = find_nikto_script()
    check_perl()
    
    # تحويل المعلمات من Python إلى صيغة Nikto
    nikto_args = []
    for arg in args:
        if arg != sys.argv[0]:  # تجاهل اسم السكربت نفسه
            nikto_args.append(arg)
    
    # عرض الشعار
    show_banner()
    
    # عرض الأمر الذي سيتم تنفيذه
    cmd = ["perl", nikto_script] + nikto_args
    print(f"{Colors.CYAN}جاري بدء الفحص...{Colors.RESET}")
    print(f"{Colors.YELLOW}الأمر:{Colors.RESET} {' '.join(cmd)}")
    print("")
    
    # تنفيذ Nikto
    try:
        process = subprocess.run(cmd, check=True)
        print("")
        print(f"{Colors.GREEN}اكتمل الفحص.{Colors.RESET}")
    except subprocess.CalledProcessError as e:
        print(f"{Colors.RED}حدث خطأ أثناء تنفيذ Nikto: {e}{Colors.RESET}")
        sys.exit(e.returncode)

def main():
    # إذا لم يتم تمرير أي معلمات أو تم طلب المساعدة
    if len(sys.argv) == 1 or "--help" in sys.argv:
        show_banner()
        show_help()
        sys.exit(0)
    
    # تمرير جميع المعلمات إلى Nikto
    run_nikto(sys.argv)

if __name__ == "__main__":
    main()
